cg/apps/lims/api.py

```diff
@@ -266,54 +266,55 @@ def _sort_by_date_run(sort_list: list):
-            processes: list[Process] = self.get_processes_from_artifacts(artifacts=artifacts)
-            for process in processes:
-                # Check which type of method document has been used
-                method_type: str = self.get_method_type(process, step_names_udfs[process_name])
-                udf_key_method_doc, udf_key_version = self.get_method_udf_values(
-                    step_names_udfs[process_name], method_type
-                )
-                methods.append(
-                    (
-                        process.date_run,
-                        self.get_method_document(process, udf_key_method_doc),
-                        self.get_method_version(process, udf_key_version),
-                        method_type,
+        try:
+            for process_name in step_names_udfs:
+                artifacts = self.get_artifacts(process_type=process_name, samplelimsid=lims_id)
+                if not artifacts:
+                    continue
+                # Get a list of parent processes for the artifacts
+                processes: list[Process] = self.get_processes_from_artifacts(artifacts=artifacts)
+                for process in processes:
+                    # Check which type of method document has been used
+                    method_type: str = self.get_method_type(process, step_names_udfs[process_name])
+                    udf_key_method_doc, udf_key_version = self.get_method_udf_values(
+                        step_names_udfs[process_name], method_type
+                    )
+                    methods.append(
+                        (
+                            process.date_run,
+                            self.get_method_document(process, udf_key_method_doc),
+                            self.get_method_version(process, udf_key_version),
+                            method_type,
+                        )
                     )
-                )
-
-        sorted_methods = self._sort_by_date_run(methods)
 
-        if sorted_methods:
-            method = sorted_methods[METHOD_INDEX]
+            sorted_methods = self._sort_by_date_run(methods)
 
-            if (
-                method[METHOD_TYPE_INDEX] == DocumentationMethod.AM
-                and method[METHOD_DOCUMENT_INDEX] is not None
-            ):
-                method_name = AM_METHODS.get(method[METHOD_DOCUMENT_INDEX])
-                return (
-                    f"{method[METHOD_DOCUMENT_INDEX]}:{method[METHOD_VERSION_INDEX]} - "
-                    f"{method_name}"
-                )
-            elif (
-                method[METHOD_TYPE_INDEX] == DocumentationMethod.ATLAS
-                and method[METHOD_DOCUMENT_INDEX] is not None
-            ):
-                return f"{method[METHOD_DOCUMENT_INDEX]} ({method[METHOD_VERSION_INDEX]})"
+            if sorted_methods:
+                method = sorted_methods[METHOD_INDEX]
 
+                if (
+                    method[METHOD_TYPE_INDEX] == DocumentationMethod.AM
```

Comment:
@AnnaLeinfelt @Karl-Svard Do we still have anything in AM?

Reply:
Added it to the issue for future refactoring: https://github.com/Clinical-Genomics/cg/issues/3163